### DISCUSSION TITLE
Downloading a file from server with JS.

### DISCUSSION THREAD

**Discussion by fabiopolimeni** [pocketbase/pocketbase#4370] [Q&A]
I am wondering if using the $send object/namespace when extending from the server with JS, it would allow to download binary files, and then upload them to the storage.

Thanks in advance.

**Comment by jf908** [comment]
I was looking for a solution to this and found out there is an API for downloading files: https://pocketbase.io/jsvm/functions/_filesystem.fileFromUrl.html.

You can then add these to the database using a [RecordUpsertForm](https://pocketbase.io/docs/js-records/#create-new-record-with-data-validations) with code like:

```js
form.addFiles('image_column', $filesystem.fileFromUrl('https://link-to-image.com/image.png'));
```
